devtools/gradle/pom.xml

```diff
@@ -62,19 +90,29 @@
                         <groupId>org.codehaus.mojo</groupId>
                         <artifactId>exec-maven-plugin</artifactId>
                         <executions>
+                            <execution>
+                                <id>gradle-clean</id>
+                                <phase>pre-clean</phase>
+                                <configuration>
+                                    <executable>${gradle.executable}</executable>
+                                    <arguments>
+                                        <argument>clean</argument>
+                                    </arguments>
+                                </configuration>
+                                <goals>
+                                    <goal>exec</goal>
+                                </goals>
+                            </execution>
                             <execution>
                                 <id>gradle</id>
                                 <phase>prepare-package</phase>
                                 <configuration>
                                     <executable>${gradle.executable}</executable>
                                     <arguments>
-                                        <argument>clean</argument>
                                         <argument>${gradle.task}</argument>
                                         <argument>-Pdescription=${project.description}</argument>
                                         <argument>-Dmaven.repo.local=${settings.localRepository}</argument>
-                                        <argument>-S</argument>
                                         <argument>--stacktrace</argument>
-                                        <argument>--no-daemon</argument>
```

Comment:
This was added way back via #11694. I'm not saying it's still 100% relevant, but IMO it would be better to not have the daemon around all the time, especially in CI?

/cc @jaikiran

Reply:
Updated the PR a little.
Local builds --> with daemon
CI builds --> no daemon

Reply:
Those 10s should be alright for CI. Better have as little differences as possible between local and CI, no?

Do you have an opinion on this @gsmet?